apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Plugin Name: Full Site Editing
+ */
+
+class WPCOM_Full_Site_Editing {
+	function __construct() {
+		$this->register_script_and_style();
+
+		add_action( 'init', [ $this, 'register_blocks' ], 100 );
+	}
+
+	function register_script_and_style() {
+		wp_register_script(
+			'wpcom-full-site-editing-script',
+			plugins_url( 'dist/full-site-editing-plugin.js', __FILE__ ),
+			[],
+			filemtime( plugin_dir_path( __FILE__ ) . 'dist/full-site-editing-plugin.js' )
+		);
+
+		wp_register_style(
+			'wpcom-full-site-editing-style',
+			plugins_url( 'dist/full-site-editing-plugin.css', __FILE__ ),
+			[],
+			filemtime( plugin_dir_path( __FILE__ ) . 'dist/full-site-editing-plugin.css' )
+		);
+	}
+
+	function register_blocks() {
+		register_block_type( 'wpcom/page-content', [
+			'editor_script' => 'wpcom-full-site-editing-script',
+			'editor_style' => 'wpcom-full-site-editing-style',
+			'render_callback' => [ $this, 'render_page_content_block' ],
+		] );
+	}
+
+	function render_page_content_block( $attributes ) {
+		if ( ! $attributes[ 'pageId' ] ) {
+			return '';
+		}
+		$post = get_post( $attributes[ 'pageId' ] );
+		setup_postdata( $post );
+		$the_content = get_the_content();
```

Comment:
I'm pretty sure we don't want `get_the_content()` here

first of all it doesn't run `the_content` filter, so we at least would favor `the_content()` instead I think
second of all, can we confirm that it runs `do_blocks()` on the content? I think it should but just in case we want to confirm by making sure something like a void block gets rendered as it should

Reply:
`do_blocks` is run within the `the_content` filter but `the_content()` prints the content so we'd either need to do some output buffering or just something like:

```php
$the_content = apply_filters( 'the_content', $post->post_content );
```